activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -2,6 +2,44 @@
 
 module ActiveRecord
   module Associations
+    module Callbacks # :nodoc:
```

Comment:
I think the proper fix for this is at the `define_callback` methods in the association builder.